File: dashatize-it/dashatize-it.py
def dashatize(num):
    if num == None:
        return "None"
    if num == 0:
        return "0"   
    if num < 0:
        num = -num
    num = str(num)
    return "".join([i if int(i)%2==0 else "-"+i+"-" for i in num]).strip("-").replace("--","-")

File: dashatize-it/dashatize-it.js
function dashatize(num) {
  if(isNaN(num)){return "NaN"}
  if(num<=0){num = -num}
  num = num.toString().split("").map(a=>a%2==0?a:"-"+a+"-").join("").replace(/--/g,"-")
  if(num[0]=="-"){num = num.slice(1)}
  if (num[num.length-1]=="-"){return num.substring(0, num.length - 1);}
  return num
};